Add documentation to fix bad hostname changes on xostor (#424)
xcp-ng/xcp-ng-org: docs/xostor/xostor.md

### How to change a hostname/node name?

> :::warning

Reviewer: What happens if the node name is not the same as the hostname? Does something break? Do we risk losing data? If there's a warning, we should be clearer about this.

The node name should _always_ be the same as the hostname. If the hostname is changed, the node name must be modified and vice versa.
:::

:::tip

> If you changed your hostname without this guide and now got errors, revert back that hostname and make sure all `*.res` files at `/var/lib/linstor.d/` on all machines of the pool are using the old hostname. Then refresh their config.

Reviewer: ```suggestion
If you changed your hostname without this guide and now got errors, revert the hostname to its previous value and make sure all `*.res` files at `/var/lib/linstor.d/` on all machines of the pool are using the old hostname. Finally, refresh their config.
```

Also, what does 'their' refer to, here?

This can be done as follow :

```bash
# On all hosts
MODIFIED_NAME='<MODIFIED_NAME>' # Hostname after changes, the one we want to revert
ORIGINAL_NAME='<ORIGINAL_NAME>' # Hostname we want to go back to

> # On incriminated host

Reviewer: ```suggestion
# On the incriminated host
```

### How to change a hostname/node name?

:::warning
The node name should _always_ be the same as the hostname. If the hostname is changed, the node name must be modified and vice versa.

> Failing to doing so will result inconsistent and outdated volumes, inability to find replica and execute certain actions on hosts (VM.start, snap…).

Reviewer: ```suggestion
If not, volumes risk being inconsistent and outdated. Replicas may not be found, and you may not be able to execute certain actions on hosts (VM.start, snap, etc.).
```

:::

:::tip

> If you changed your hostname without this guide and now got errors, revert the hostname to it's previous value and make sure all `*.res` files at `/var/lib/linstor.d/` on all machines of the pool are using the old hostname. Then reload linstor config on all machines.

Reviewer: ```suggestion
If you changed your hostname without this guide and now got errors, revert the hostname to its previous value and make sure all `*.res` files at `/var/lib/linstor.d/` on all machines of the pool are using the old hostname. Then reload linstor config on all machines.
```
